src/adts/bit_array_impl.h

```diff
@@ -361,7 +361,7 @@ bit_array_append_bucket(BitArray *array, uint8 bits_used, uint64 bucket)
 static uint64
 bit_array_low_bits_mask(uint8 bits_used)
 {
-	if (bits_used == 64)
+	if (bits_used >= CHAR_BIT * sizeof(UINT64CONST(1)))
```

Comment:
I guess we could only change the comparison from >= to here, since we are using the likely construction below anyway, it is up to you.

Reply:
in this case I think `>= 64` is the correct construction, how large `uint64` is normally stored in memory as is irrelevant (granted, if it's anything other than 64 bits, a whole lot of other things will break, but I don't think that's a reason to make this code less clear).
Incidentally, if we do use `sizeof`, shouldn't it just be `CHAR_BIT *  sizeof(uint64)`?

Reply:
Since it is undefined if the shift exceeds the number of bits I only bothered about that case. Since we are shifting `UINT64CONST(1)` I am using the same expression for the sizeof. Then I don't have to bother about the exact type. The result will be the same anyway: I can change.

Reply:
I'm using 64 right now, but strictly speaking how large `uint64` is in memory is very relevant since that decides the number of bits, and a shift of same or large than the number of bits of the type will lead to undefined behavior.